tst/EnergyPlus/unit/UnitarySystem.unit.cc

```diff
@@ -17185,3 +17185,747 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_DesuperHeatCoilStptNodeTest)
+    EXPECT_EQ(thisSys.m_CoolingCoilType_Num, DataHVACGlobals::Coil_CoolingWater);
+    EXPECT_EQ(thisSys.m_CoolingCoilName, "WATER COOLING COIL");
+    EXPECT_TRUE(thisSys.m_CoolCoilExists);
+    EXPECT_TRUE(thisSys.m_waterSideEconomizerFlag);
+    EXPECT_TRUE(thisSys.m_RunOnSensibleLoad);
+    EXPECT_FALSE(thisSys.m_RunOnLatentLoad);
+    // Default control status
+    EXPECT_FALSE(thisSys.runWaterSideEconomizer);
+    EXPECT_EQ(thisSys.WaterSideEconomizerStatus, 0);
+    // set some flags
+    thisSys.m_MyFanFlag = false;
+    state->dataGlobal->SysSizingCalc = true;
+    FirstHVACIteration = true;
+    bool HXUnitOn(false);
+    int CompOn(0);
+    int ZoneOAUnitNum(0);
+    Real64 OAUCoilOutTemp(0.0);
+    // initial assumptions
+    thisSys.m_DesiredOutletHumRat = 0.10;
+    state->dataLoopNodes->Node(thisSys.m_SystemCoolControlNodeNum).TempSetPoint = 10.0;
+    thisSys.m_DesiredOutletTemp = state->dataLoopNodes->Node(thisSys.m_SystemCoolControlNodeNum).TempSetPoint;
+    state->dataWaterCoils->WaterCoil(1).InletWaterMassFlowRate = ColdWaterMassFlowRate;
+    state->dataWaterCoils->WaterCoil(1).MaxWaterMassFlowRate = ColdWaterMassFlowRate;
+    state->dataWaterCoils->WaterCoil(1).DesAirVolFlowRate = 1.0;
+    state->dataLoopNodes->Node(state->dataWaterCoils->WaterCoil(1).WaterInletNodeNum).MassFlowRate = ColdWaterMassFlowRate;
+    ;
+    // Test 1: economizer inlet water temperature is favorable, expect coil ON
+    // run init and check the coil system operating condition and control status
+    thisSys.initUnitarySystems(*state, AirLoopNum, FirstHVACIteration, ZoneOAUnitNum, OAUCoilOutTemp);
+    EXPECT_TRUE(thisSys.runWaterSideEconomizer);
+    EXPECT_EQ(thisSys.WaterSideEconomizerStatus, 1);
+    thisSys.controlCoolingSystemToSP(*state, AirLoopNum, false, HXUnitOn, CompOn);
+    EXPECT_EQ(thisSys.m_CoolingPartLoadFrac, 1.0);
+    thisSys.calcUnitaryCoolingSystem(*state, AirLoopNum, false, thisSys.m_CoolingPartLoadFrac, CompOn, OnOffAirFlowRatio, CoilCoolHeatRat, false);
+    EXPECT_NEAR(19267.0, state->dataWaterCoils->WaterCoil(1).TotWaterCoolingCoilRate, 1.0);
+    ;
+    // Test 2: economizer inlet water temperature is NOT favorable, expect coil OFF
+    state->dataWaterCoils->WaterCoil(1).InletWaterTemp = 29.5;
+    state->dataLoopNodes->Node(state->dataWaterCoils->WaterCoil(1).WaterInletNodeNum).Temp = state->dataWaterCoils->WaterCoil(1).InletWaterTemp;
+    thisSys.initUnitarySystems(*state, AirLoopNum, FirstHVACIteration, ZoneOAUnitNum, OAUCoilOutTemp);
+    EXPECT_FALSE(thisSys.runWaterSideEconomizer);
+    EXPECT_EQ(thisSys.WaterSideEconomizerStatus, 0);
+    thisSys.controlCoolingSystemToSP(*state, AirLoopNum, false, HXUnitOn, CompOn);
+    EXPECT_EQ(thisSys.m_CoolingPartLoadFrac, 0.0);
+    thisSys.calcUnitaryCoolingSystem(*state, AirLoopNum, false, thisSys.m_CoolingPartLoadFrac, CompOn, OnOffAirFlowRatio, CoilCoolHeatRat, false);
+    EXPECT_NEAR(0.0, state->dataWaterCoils->WaterCoil(1).TotWaterCoolingCoilRate, 0.001);
+}
```

Comment:
Unit test for "COILSYSTEM:COOLING:WATER" calculation with the coil economizer control status ON and OFF.